### PR TITLE
Cleanup outer and add GEMM routing for CUDA

### PR DESCRIPTION
**Pull request by kshyatt**
# Description
This actually turned into a code addition. Added some functions as fall backs for array types that `@strided` doesn't work on, and fixed up some `ITensorGPU` code to be more consistent with `ITensors`.

Fixes # (issue)

# How Has This Been Tested?

`julia test/runtests.jl` passed for `ITensorGPU`, `NDTensors`, and `ITensor`.

# Checklist:

- [x] My code follows the style guidelines of this project. Please run `using JuliaFormatter; format(".")` in the base directory of the repository (`~/.julia/dev/ITensors`) to format your code according to our style guidelines.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

File: ITensorGPU/src/cuitensor.jl [hunk 1]
```diff
@@ -14,6 +18,41 @@ function cuITensor(x::S, inds::IndexSet{N}) where {S<:Number,N}
...
+
+function ITensor(
+  as::AliasStyle,
+  eltype::Type{<:Number},
+  A::CuArray{<:Number},
+  inds::Indices{Index{Int}};
+  kwargs...,
+)
+  @show size(A)
+  @show dim(inds)
```
Comment (emstoudenmire) [inline ITensorGPU/src/cuitensor.jl:30]: Looks like there are some @show calls here that need to be commented out?

**Comment by mtfishman** [comment]
Thanks @kshyatt, I'll take a look.

Could you run the formatter? You can do:
```julia
julia> cd(joinpath(ENV["HOME"], ".julia", "dev", "ITensors"))

julia> using JuliaFormatter

julia> format(".")
```

**Comment by emstoudenmire** [comment]
Or if it's handy to either of you, I made a little script that I run from the top level of the folders containing files I want to format. Here's the script:

```
#!/bin/bash
julia --banner=no --color=yes --startup-file=no -e "using JuliaFormatter; format(\".\";verbose=true)"
```
then I put that into a file in my path called `julia_format`, chmod it to 755, and run it as a command.

**Comment by mtfishman** [comment]
Looks good Katie, thanks! Looks like that ITensor constructor you added could get turned into a generic constructor for `AbstractArray` (i.e. both `CuArray` or `Array`). I'm hoping we can start moving in the direction of making the code in ITensors.jl more generic so it work better with `CuArray` data types.

I think the only remaining thing is removing the `@show` lines that Miles pointed out, then I'll merge if you think this is good to go.

**Review by mtfishman — APPROVED** [review]

**Pull request merged**